### PR TITLE
server: add error HTML templates with error description.

### LINKED ISSUE #657: server: error HTML templates
Right now they're just plain text.

### PR DESCRIPTION
**Pull request by rithujohn191**
Tested the change with the example-app, it successfully displays an error template with error type and error description.

closes #657 

@ericchiang

### CODE REVIEW

File: server/handlers.go [hunk 1]
```diff
@@ -294,14 +294,14 @@ func (s *Server) handleConnectorCallback(w http.ResponseWriter, r *http.Request)
 	identity, err := callbackConnector.HandleCallback(parseScopes(authReq.Scopes), r)
 	if err != nil {
 		s.logger.Errorf("Failed to authenticate: %v", err)
-		s.renderError(w, http.StatusInternalServerError, errServerError, "")
+		s.renderError(w, http.StatusInternalServerError, errServerError, "Connector callback error")
```
Comment (rithujohn191) [inline server/handlers.go]: changed it to "Login error"

**Comment by rithujohn191** [comment]
approval

File: server/handlers.go [hunk 2]
```diff
@@ -138,13 +138,13 @@ func (s *Server) discoveryHandler() (http.HandlerFunc, error) {
 func (s *Server) handleAuthorization(w http.ResponseWriter, r *http.Request) {
 	authReq, err := s.parseAuthorizationRequest(s.supportedResponseTypes, r)
 	if err != nil {
-		s.renderError(w, http.StatusInternalServerError, err.Type, err.Description)
+		s.renderError(w, http.StatusInternalServerError, errServerError, "Failed to parse authorization request")
```
Comment (rithujohn191) [inline server/handlers.go]: A lot of the existing http errors actually mentioned "authorization request". I'll clean those up too then.

**Comment by rithujohn191** [comment]
approval 1

Improved "error type" message that gets displayed.

**Comment by ericchiang** [comment]
lgtm. we can continue improving the exact error messages later.

**Pull request merged**